ixmp4/data/abstract/iamc/timeseries.py

```diff
@@ -22,6 +29,21 @@ def __str__(self) -> str:
 ModelType = TypeVar("ModelType", bound=TimeSeries)
 
 
+class EnumerateKwargs(TypedDict, total=False):
+    id: int
+    id__in: Iterable[int]
+    region: dict[str, int | str | Iterable[int] | Iterable[str]]
+    unit: dict[str, int | str | Iterable[int] | Iterable[str]]
+    variable: dict[str, int | str | Iterable[int] | Iterable[str]]
+    run: dict[
```

Comment:
Now that im seeing another one of these (this time for `run`): If we can somehow generate these `TypedDict` classes it should theoretically be possible to type this field with another one of those?

Reply:
See comment above, lets not go that far!